File: data/src/main/java/com/tonitealive/app/data/UiThread.java
package com.tonitealive.app.data;

import com.tonitealive.app.domain.executor.PostExecutionThread;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;


public class UiThread implements PostExecutionThread {

    @Override
    public Scheduler getScheduler() {
        return AndroidSchedulers.mainThread();
    }

}

File: data/src/main/kotlin/com/tonitealive/app/data/UiThread.kt
package com.tonitealive.app.data

import com.tonitealive.app.domain.executor.PostExecutionThread
import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers


class UiThread : PostExecutionThread {
    override val scheduler: Scheduler
        get() = AndroidSchedulers.mainThread()
}